src/DataSets/log.js

```diff
@@ -1,44 +1,73 @@
 
+const maxLogsPerPage = 200;  // TODO: maybe make it readable from the dataStore
+const maxLogPages = 100;
 
-async function log(actionName, status, dataset) {
+
+async function log(actionName, status, datasets) {
     // Log the name of the action that has been executed, its status
     // ("success", "failed"), by whom and on which datasets.
-    const maxLogs = 1e3;
-
-    const d2 = await getD2();
-    const store = await d2.dataStore.get('dataset-configuration');
-
-    const logIndex = await store.get('logIndex').catch(() => 0);
-    const logs = await store.get('logs').catch(() => []);
-
-    const datasets = Array.isArray(dataset) ? dataset : [dataset];
+    const store = await getStore();
+    const logs = await getLogs([0], store);
+    logs.push(await logEntry(actionName, status, datasets));
+    setLogs(logs, store);
+}
 
-    const newLog = {
+async function logEntry(actionName, status, datasets) {
+    // Return an object (a "dictionary") with the information we want to log.
+    datasets = Array.isArray(datasets) ? datasets : [datasets];
+    const user = await getD2().then(d2 => d2.currentUser);
+    return {
         date: new Date().toISOString(),
         action: actionName,
         status: status,
-        user: {displayName: d2.currentUser.name,
-               username: d2.currentUser.username,
-               id: d2.currentUser.id},
-        datasets: datasets.map(ds => ({displayName: ds.name, id: ds.id})),
+        user: {
+            displayName: user.name,
+            username: user.username,
+            id: user.id,
+        },
+        datasets: datasets.map(ds => ({
+            displayName: ds.name,
+            id: ds.id,
+        })),
     };
+}
+
+async function getLogs(pages, store) {
+    // Return the concatenated logs for the given pages (relative to
+    // logsPageCurrent) if they exist, or an empty list otherwise.
+    pages = pages || [-1, 0];
+    store = store ? store : await getStore();
+    const logsPageCurrent = await get('logs-page-current', 0, store);
+    const pagesNames = pages.map(n => 'logs-page-' + mod(logsPageCurrent + n, maxLogPages));
+    const logs = await Promise.all(pagesNames.map(name => get(name, [], store)));
+    return [].concat(...logs);
+}
 
-    if (logs.length < maxLogs)
-        logs.push(newLog)
-    else
-        logs[logIndex] = newLog;
+async function setLogs(logs, store) {
+    // Save the given list of logs in the current page index and
+    // increase it if necessary.
+    const logsPageCurrent = await get('logs-page-current', 0, store);
+    store.set('logs-page-' + logsPageCurrent, logs);
+    if (logs.length >= maxLogsPerPage)
+        store.set('logs-page-current', mod(logsPageCurrent + 1, maxLogPages));
+}
 
-    store.set('logIndex', (logIndex + 1) % maxLogs);
-    store.set('logs', logs);
+async function get(key, defaultValue, store) {
+    // Return the value for key in the store.
+    return await store.get(key).catch(() => defaultValue);
```

Comment:
Not a problem, but in simple functions like this, the `async`/`await` can be removed altogether.

Reply:
You are right, it's unneeded and I was just being a bit paranoid. Removed (the simpler the better), thanks!